lib/utils.coffee

```diff
@@ -274,10 +276,16 @@ exports.getBody = processBody = (response) ->
 				return response.buffer()
 			throw new Error('This `fetch` implementation does not support decoding binary streams.')
 
-		if includes(contentType, 'application/json')
+		if responseFormat is 'json' or (not responseFormat? and includes(contentType, 'application/json'))
 			return response.json()
 
-		return response.text()
+		if responseFormat is 'none'
+			return Promise.resolve(null)
+
+		if not responseFormat? or responseFormat is 'text'
+			return response.text()
+
+		return Promise.reject(new Error("Unknown responseFormat: #{responseFormat}"))
```

Comment:
We now have a `ResinInvalidParameterError`, which would fit here nicely: https://github.com/resin-io-modules/resin-errors/blob/master/lib/errors.coffee#L234